File: JPARecipes/src/main/java/com/skilldistillery/recipes/entities/Recipe.java
package com.skilldistillery.recipes.entities;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Objects;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;

@Entity
public class Recipe {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;
	private String title;
	private String description;
	private Integer servings;
	@Column(name="prep_time_minutes")
	private int prepTime;
	@Column(name="cook_time_minutes")
	private int cookTime;
	private String directions;
	private String source;
	@Column(name="website_url")
	private String websiteURL;
	@Column(name="image_url")
	private String imageURL;
	@Column(name="create_date")
	@CreationTimestamp
	private LocalDateTime createDate;
	@Column(name="last_update")
	@UpdateTimestamp
	private LocalDateTime lastUpdate;
	private Boolean enabled;
	@ManyToOne
	@JoinColumn(name="food_type_id")
	private FoodType foodType;
	@JsonIgnore
	@OneToMany(mappedBy="recipe")
	private List<Review> reviews;
	@ManyToMany(mappedBy="recipes")
	private List<Category> categories;
	@ManyToMany(mappedBy="recipes")
	private List<Ingredient> ingredients;
	
	public Recipe() {
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public int getPrepTime() {
		return prepTime;
	}

	public void setPrepTime(int prepTime) {
		this.prepTime = prepTime;
	}

	public int getCookTime() {
		return cookTime;
	}

	public void setCookTime(int cookTime) {
		this.cookTime = cookTime;
	}

	public Integer getServings() {
		return servings;
	}

	public void setServings(Integer servings) {
		this.servings = servings;
	}

	public String getDirections() {
		return directions;
	}

	public void setDirections(String directions) {
		this.directions = directions;
	}

	public String getSource() {
		return source;
	}

	public void setSource(String source) {
		this.source = source;
	}

	public String getWebsiteURL() {
		return websiteURL;
	}

	public void setWebsiteURL(String websiteURL) {
		this.websiteURL = websiteURL;
	}

	public String getImageURL() {
		return imageURL;
	}

	public void setImageURL(String imageURL) {
		this.imageURL = imageURL;
	}

	public LocalDateTime getCreateDate() {
		return createDate;
	}

	public void setCreateDate(LocalDateTime createDate) {
		this.createDate = createDate;
	}

	public LocalDateTime getLastUpdate() {
		return lastUpdate;
	}

	public void setLastUpdate(LocalDateTime lastUpdate) {
		this.lastUpdate = lastUpdate;
	}

	public Boolean getEnabled() {
		return enabled;
	}

	public void setEnabled(Boolean enabled) {
		this.enabled = enabled;
	}

	public FoodType getFoodType() {
		return foodType;
	}

	public void setFoodType(FoodType foodType) {
		this.foodType = foodType;
	}

	public List<Review> getReviews() {
		return reviews;
	}
	
	public List<Category> getCategories() {
		return categories;
	}

	public void setCategories(List<Category> categories) {
		this.categories = categories;
	}

	public void setReviews(List<Review> reviews) {
		this.reviews = reviews;
	}

	public List<Ingredient> getIngredients() {
		return ingredients;
	}

	public void setIngredients(List<Ingredient> ingredients) {
		this.ingredients = ingredients;
	}

	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Recipe other = (Recipe) obj;
		return id == other.id;
	}

	@Override
	public String toString() {
		return "Recipe [id=" + id + ", title=" + title + ", description=" + description + ", servings=" + servings
				+ ", prepTime=" + prepTime + ", cookTime=" + cookTime + ", directions=" + directions + ", source="
				+ source + ", websiteURL=" + websiteURL + ", imageURL=" + imageURL + ", createDate=" + createDate
				+ ", lastUpdate=" + lastUpdate + ", enabled=" + enabled + "]";
	}
	
	
}

File: ngRefriedRecipes/src/app/models/recipe.ts
import { Category } from "./category";
import { FoodType } from "./food-type";
import { RecipeIngredient } from "./recipe-ingredient";
import { Review } from "./review";

export class Recipe {
  id: number;
  title: string;
  description: string;
  servings: string;
  prepTime: number;
  cookTime: number;
  totalTime: number;
  directions: string;
  source: string;
  websiteUrl: string;
  imageUrl: string;
  enabled: boolean;
  createDate: string;
  lastUpdate: string;
  foodType: FoodType;
  reviews: Review [] | undefined;
  categories: Category [] | undefined;
  recipeIngredients: RecipeIngredient[] | undefined;

constructor(
  id: number = 0,
  title: string = '',
  description: string = '',
  servings: string = ' ',
  prepTime: number = 0,
  cookTime: number = 0,
  totalTime: number = 0,
  directions: string = '',
  source: string = ' ',
  websiteUrl: string = '',
  imageUrl: string = ' ',
  enabled: boolean = true,
  createDate: string = '',
  lastUpdate: string = '',
  foodType: FoodType = new FoodType(),
  reviews: Review[] = [],
  categories: Category[] = [],
  recipeIngredients: RecipeIngredient[] = [],
)
{
this.id = id;
this.title = title;
this.description = description;
this.servings = servings;
this.prepTime = prepTime;
this.cookTime = cookTime;
this.totalTime = totalTime;
this.directions = directions;
this.source = source;
this.websiteUrl = websiteUrl;
this.imageUrl =  imageUrl;
this.createDate = createDate;
this.lastUpdate = lastUpdate;
this.enabled = enabled;
this.foodType = foodType;
this.reviews = reviews;
this.categories = categories;
this.recipeIngredients = recipeIngredients;
}

}
